### ISSUE TITLE
discussion: dom.storage.next_gen

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [arkenfox/user.js#1059]
I think it's time to add and flip this actively true in the user.js - especially now ESR68 is EOL.

I've been using it for probably well over a year. #658 is where it was came up, back in April 2019 (FF66 was stable back then), and I had quite a few emails with Johann Hofmann and did shitloads of testing digging into that issue (not everyone could replicate, cuz .. wonkiness). The hope then was to get it into ESR68. Well that didn't happen. Then it was ESR78 .. that ain't happened either. It has been default true in Nightly for ages (a year + I think, it's just never ridden the train)

Generally speaking ... `dom.storage.next_gen` (LSNG) is an improvement: it makes the whole quota/storage framework thingie more robust with less wonky arbitrary asynchronous timing mishaps etc
- e.g.: a service worker across two tabs can now see each other, etc
- e.g. it improves the storage inspector
- e.g. it helps insure localStorage is sanitized on close : see #658 (among others)

Here's some current tickets
- [1599979](https://bugzilla.mozilla.org/show_bug.cgi?id=1599979) [META] Enable LSNG on Release
- [1482662](https://bugzilla.mozilla.org/show_bug.cgi?id=1482662) [META] The War on Storage Initialization Failures

There are probably better tickets to explain why it hasn't been flipped after 18 months. My understanding is it's a very small % of failures (old data, old data structure?). And since we sanitize most data on close, this shouldn't really be an issue for our users

I've also never seen it have any issues with extension data, but I have a nagging thought in the back of my mind that uM/uBO mentioned it a few times way back: not sure if it was good or bad (may be the same issue of old data/structure? in other words initialization) - but we're well into FF78+ now

I see @gwarser just lodged a [ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1675643) , but that's about web ext's API, and should be resolved before stable - but again, just another reason we should flip this

tl;dr: benefits = super good, potential pitfalls = practically zero (if you have an initialization issue, clear your data and start again AFAICT)

Class, discuss!

**Comment by Synkky** [comment]
I've been using it for ages too, it's that kind of thing you just flip and forget tbh, never gave me any problems either.

**Comment by Thorin-Oakenpants** [comment]
FYI: here's the original ticket: https://bugzilla.mozilla.org/286798 FF65+
